src/scripting/game_lua_kernel.cpp

```diff
@@ -3900,22 +3900,35 @@ static std::string read_event_name(lua_State* L, int idx)
 	}
 }
 
+static int read_opt_event_priority(lua_State* L, int idx)
```

Comment:
I don't get why this needs to be a separate function…? It's only used once…

Reply:
I didn't find that API. Sounds like just what is needed.

Reply:
There was a reason why I couldn't find it. That API has been deprecated since Lua 5.3, and apparently Wesnoth's compilation flags don't include it.

Reply:
It was renamed to `luaL_optinteger`? Since that's what you ended up using (until changing it to float that is).

Reply:
It was removed in favor of ``luaL_optinteger``, which uses the largest available integer type.